frappe/email/inbox.py

```diff
@@ -161,20 +161,45 @@ def make_lead_from_communication(communication, ignore_communication_links=False
 
 @frappe.whitelist()
 def make_opportunity_from_communication(communication, ignore_communication_links=False):
```

Comment:
can we move this to ERPNext via a hook?

Reply:
Although I fully agree with you, unfortunately, re-writing the Inbox is going to require more heavy-lifting then I feel comfortable with and can afford to at this time. 

The intention of this PR is more of a bugfix than an enhancement, or the proper refactor it deserves. It is possible that in the future I may be able to do so, but unfortunately, it won't be the immediate future.